src/BenGorUser/User/Domain/Model/User.php

```diff
@@ -192,9 +189,17 @@ public function id()
 
     /**
      * Accepts the invitation request.
+     *
+     * @throws UserTokenExpiredException when the token is expired
      */
     public function acceptInvitation()
     {
+        if (null === $this->isInvitationTokenExpired()) {
```

Comment:
And maybe we can consider that when you accept an invitation token it expires. The use case will be as follows for `isInvitationTokenExpired()`:
* User has invitation token within expire limit: returns `false`
* User has invitation token out of expire limit: returns `true`
* User has no invitation token: returns `true`